lib/graphql/argument.rb

```diff
@@ -96,15 +96,19 @@ def prepare=(prepare_proc)
 
     NO_DEFAULT_VALUE = Object.new
     # @api private
-    def self.from_dsl(name, type = nil, description = nil, default_value: NO_DEFAULT_VALUE, as: nil, prepare: DefaultPrepare, **kwargs, &block)
+    def self.from_dsl(name, type_or_argument = nil, description = nil, default_value: NO_DEFAULT_VALUE, as: nil, prepare: DefaultPrepare, **kwargs, &block)
+      if type_or_argument.is_a?(GraphQL::Argument)
+        return type_or_argument.redefine(name: name.to_s)
+      end
+
       argument = if block_given?
```

Comment:
should we handle a block when provided an existing arg? Redefine attributes in the block 🤔 ? That might be confusing.

Reply:
It might come in handy, but I'm happy to leave it for later. 

We [do it for fields](https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/define/assign_object_field.rb#L37-L41) but it makes more sense there because some things (like `argument`) _must_ be defined in the block.